### DISCUSSION TITLE
How to contribute?

### DISCUSSION THREAD

**Discussion by Bleb1k** [bevyengine/bevy#5492] [Q&A] [answered]
Do i need to create a branch or create a fork, or something else?

I created resource `<Input<ScanCode>>` and system that updates it (and this system will be more compact if directly injected into bevy's `keyboard_input_system`)
This is done because i want to have controls independent of keyboard layout and this feels too important to not be in the engine so i want to contribute!

Thanks!

**Comment by Nilirad** [comment, marked as answer]
Hi, check [this section] of `CONTRIBUTING.md`

[this section]: https://github.com/bevyengine/bevy/blob/main/CONTRIBUTING.md#contributing-code

**Comment by Bleb1k** [comment]
Thanks alot!!
